### ISSUE TITLE
aws sam local start-api regression on "provided" runtime missing header "lambda-runtime-invoked-function-arn"

### ISSUE THREAD

**Issue by time4tea** [aws/aws-sam-cli#2715]
### Description:

Using AWS sam 1.3.2, runtime provided, it is possible to run a rustlang AWS api using sam local with runtime: provided. Under newer versions, the local versions of the deployment fail: (edit: the deployment itself succeeds, but the running lambda is not usable)

```
main' panicked at 'no entry found for key "lambda-runtime-invoked-function-arn"', /home/richja/.cargo/registry/src/github.com-1ecc6299db9ec823/lambda_runtime-0.3.0/src/types.rs2021-03-15 09:24:50,041 TRACE [want] signal found waiting giver, notifying
```

this seems to be triggered in : lambda-runtime-0.3.0, src/types.rs

```
            invoked_function_arn: headers["lambda-runtime-invoked-function-arn"]
                .to_str()
                .expect("Missing arn; this is a bug")
                .to_owned(),
```

### Steps to reproduce:

Make a rust lambda, using sam 1.3.2 - it will work.
Make a rust lambda, using sam 1.19 - it will work when deployed into AWS, but not locally, failing with above panic.

### Observed result:
panic in lambda_runtime
```
  15:     0x7f78e1fb6139 - <lambda_runtime::types::Context as core::convert::TryFrom<http::header::map::HeaderMap>>::try_from::h0ca51065315b7aa3
  16:     0x7f78e1f38cb8 - lambda_runtime::run::{{closure}}::h444da055591230ba
  17:     0x7f78e1f44786 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h88d3f3d579db64e0
  18:     0x7f78e1f3b47b - std::thread::local::LocalKey<T>::with::h09bc194ffe5b62c8
  19:     0x7f78e1f9ec8d - tokio::park::thread::CachedParkThread::block_on::h230b34efeab4dda5
  20:     0x7f78e1f8e494 - tokio::runtime::thread_pool::ThreadPool::block_on::h4aed425d9cb7bbeb
  21:     0x7f78e1f5f5b8 - tokio::runtime::Runtime::block_on::h00f620b9f5b57519
```

this happens always, with any input

example template:

```
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'

Resources:
  HelloRustFunction:
    Type: AWS::Serverless::Function
    Metadata:
      BuildMethod: makefile
    Properties:
      FunctionName: HelloRust
      Handler: bootstrap.is.real.handler
      CodeUri: .
      Runtime: provided
      Timeout: 300
      MemorySize: 2048
      Environment:
        Variables:
          RUST_BACKTRACE: full
      Events:
        RootEvent:
          Type: Api
          Properties:
            Path: /
            Method: any
        GatewayEvent:
          Type: Api
          Properties:
            Path: /{proxy+}
            Method: any
```

**Comment by Fishrock123** [comment]
Hi, this makes the Rust aws runtime impossible to use with SAM, as it crashes, which has been deemed to be a bug here. See https://github.com/awslabs/aws-lambda-rust-runtime/pull/322

**Comment by time4tea** [comment]
Just wanted to make it clear that it is a problem with the runtime, not a problem with rust or the rust/lambda interface. The runtime must provide the header as documented in

https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html

This view is strengthened due to the fact that this only happens on sam local, not on real lambda, where the header is correctly supplied.

**Comment by brainstorm** [comment]
I believe this has been fixed in https://github.com/awslabs/aws-lambda-rust-runtime/pull/332?

**Comment by ssenchenko** [comment]
Closing as it's partially fixed in [#332](https://github.com/awslabs/aws-lambda-rust-runtime/pull/332) and partially a duplicate of [#3132](https://github.com/aws/aws-sam-cli/issues/3132)
